anta/tests/logging.py

```diff
@@ -0,0 +1,297 @@
+"""
+Test functions related to the EOS various logging settings
+"""
+from __future__ import annotations
+
+import logging
+import re
+from typing import List, Optional
+
+from anta.models import AntaTest, AntaTestCommand
+
+logger = logging.getLogger(__name__)
+
+
+class VerifyLoggingPersistent(AntaTest):
+    """
+    Verifies if logging persistent is enabled and logs are saved in flash.
+
+    Expected Results:
+        * success: The test will pass if logging persistent is enabled and logs are in flash.
+        * failure: The test will fail if logging persistent is disabled or no logs are saved in flash.
+    """
+
+    name = "VerifyLoggingPersistent"
+    description = "Verifies if logging persistent is enabled and logs are saved in flash."
+    categories = ["logging"]
+    commands = [
+        AntaTestCommand(command="show logging | awk 'NR==4 {print; exit}'", ofmt="text"),
+        AntaTestCommand(command="dir flash:/persist/messages", ofmt="text"),
+    ]
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        """
+        Run VerifyLoggingPersistent validation.
+        """
+        self.logger.debug(f"self.instance_commands is: {self.instance_commands}")
+        log_output = self.instance_commands[0].output
```

Comment:
When you have `ofmt="text"` cast to `str`
```suggestion
        log_output = cast(str, self.instance_commands[0].output)
```